src/wallet.ts

```diff
@@ -243,6 +244,7 @@ WalletMiddlewareOptions): JsonRpcMiddleware<any, Block> {
 
     const address = await validateAndNormalizeKeyholder(params[0], req);
     const message = normalizeTypedMessage(params[1]);
+    validatePrimaryType(message);
```

Comment:
Do we not want to validate this before signing process? Like we did it in the `No PrimaryType defined` example in the test dapp?

If this is a malformed signature and not applying the rules of `EIP-712` I don't think we should even show the request in the extension, must be rejected immediately.

Reply:
For example if we change `TYPED_MESSAGE_SCHEMA` in this package to 

```
{
  "type": "object",
  "properties": {
    "types": {
      "type": "object",
      "additionalProperties": {
        "type": "array",
        "items": {
          "type": "object",
          "properties": {
            "name": { "type": "string" },
            "type": { "type": "string" }
          },
          "required": ["name", "type"]
        }
      }
    },
    "primaryType": { "type": "string" },
    "domain": { "type": "object" },
    "message": { "type": "object" }
  },
  "required": ["types", "primaryType", "domain", "message"],
  "dependencies": {
    "primaryType": {
      "properties": {
        "types": {
          "type": "object",
          "required": {
            "$data": "1/primaryType"
          }
        }
      }
    }
  }
}
```

This schema will be validated before signature is getting added into controller. (in `signature-controller`)

Reply:
Here we are doing same, rejecting before signing process.

Reply:
Indeed, in this PR the validation is happening before the signing process..

Though I do wonder whether we should be using a schema instead of manually validating each expectation we have.

Reply:
I am fine with both approaches, just wanted to highlight that schema validation is also happening today. On both approaches, I think it make sense to centralise them in one place, do it on the middleware functions make sense as this PR do

Reply:
I agree @OGPoyraz , but schema validation is currently happening in SignatureController, may be we move it here.

Reply:
Exactly, there are multiple tasks in the backlog for that, we will handle that next year